server/workflow/workflow_server.go

```diff
@@ -129,7 +129,50 @@ func (s *workflowServer) GetWorkflow(ctx context.Context, req *workflowpkg.Workf
 	return wf, nil
 }
 
-func mergeWithArchivedWorkflows(liveWfs v1alpha1.WorkflowList, archivedWfs v1alpha1.WorkflowList, numWfsToKeep int) *v1alpha1.WorkflowList {
+func cursorPaginationByResourceVersion(items []v1alpha1.Workflow, resourceVersion string, limit int64, wfList *v1alpha1.WorkflowList) {
+	// Use Kubernetes resourceVersion for cursor pagination.
+	// Sort the Kubernetes results in descending order by resourceVersion.
+	// To implement cursor pagination with filtering based on resourceVersion, start by sorting in descending order according to the resourceVersion.
+	sort.Slice(items, func(i, j int) bool {
+		itemIRV, _ := strconv.Atoi(items[i].ResourceVersion)
+		itemJRV, _ := strconv.Atoi(items[j].ResourceVersion)
+		return itemIRV > itemJRV
+	})
+
+	// Due to the descending sorting above, the items are filtered to have a resourceVersion smaller than the received value.
+	// The data with values smaller than the received resourceVersion on the current page will be used for the next page.
+	if resourceVersion != "" {
+		var newItems []v1alpha1.Workflow
+		for _, item := range items {
+			targetRV, _ := strconv.Atoi(item.ResourceVersion)
+			receivedRV, _ := strconv.Atoi(resourceVersion)
+			if targetRV < receivedRV {
+				newItems = append(newItems, item)
+			}
+			items = newItems
+		}
+	}
+
+	// Indexing list by limit count
+	if limit != 0 {
+		endIndex := int(limit)
+		if endIndex > len(items) || limit == 0 {
+			endIndex = len(items)
+		}
+		wfList.Items = items[0:endIndex]
+	} else {
+		wfList.Items = items
+	}
+
+	// Calculate new offset for next batch
```

Comment:
For next page?

Reply:
Yes, for next page.
to clarify I modify the word `batch` to `page`

d433216c5e6739ccd5cc17aa1d006019af2b431a